File: Data/DanhGia.cs
namespace WebsiteSmartHome.Data;

public partial class DanhGia
{
    public Guid Id { get; set; }

    public Guid MaDonHang { get; set; }

    public Guid MaSanPham { get; set; }

    public int SoSao { get; set; }

    public string? NoiDung { get; set; }

    public System.DateTime? NgayDanhGia { get; set; }

    public virtual ChiTietDonHang ChiTietDonHang { get; set; } = null!;
}

File: Frontend/admindoan/src/types/danhgia.ts
export interface DanhGiaDto {
    id?: string;
    maDonHang: string;
    maSanPham: string;
    maNguoiDung: string;
    tenNguoiDung: string;
    tenSanPham: string;
    soSao: number;
    noiDung: string;
    ngayDanhGia: string;
    ngayDatHang: string;
  }
  
  export interface CreateDanhGiaDto {
    maDonHang: string;
    maSanPham: string;
    maNguoiDung: string;
    soSao: number;
    noiDung: string;
  }
  
  export interface UpdateDanhGiaDto {
    soSao: number;
    noiDung: string;
  }
  
  export interface DanhGiaDetailDto {
    id: string;
    maDonHang: string;
    maSanPham: string;
    maNguoiDung: string;
    soSao: number;
    noiDung: string;
    ngayDanhGia: string;
    tenNguoiDung: string;
    tenSanPham: string;
    ngayDatHang: string | null;
  }
  
  export interface PagedResponse<T> {
    statusCode: number;
    Data: T[];
    PageNumber: number;
    PageSize: number;
    TotalPages: number;
    TotalCount: number;
  }
  